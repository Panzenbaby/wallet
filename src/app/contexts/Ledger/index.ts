export * from "./contracts";
export * from "./hooks/connection";
export * from "./hooks/scanner";
export * from "./Ledger";
