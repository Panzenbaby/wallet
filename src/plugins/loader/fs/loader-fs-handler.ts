/* istanbul ignore file */

import { ipcMain } from "electron";
import os from "os";
import path from "path";

import { PluginLoaderFileSystem } from "./loader-fs";

export const injectHandler = () => {
	const root = process.env.PLUGINS_DIR || path.resolve(os.homedir(), ".payvo-wallet", "plugins");

	const finder = new PluginLoaderFileSystem(root);

	ipcMain.handle("plugin:loader-fs.search", (_, profileId: string) => finder.search(profileId));
	ipcMain.handle("plugin:loader-fs.remove", (_, dir: string) => finder.remove(dir));
	ipcMain.handle("plugin:loader-fs.find", (_, dir: string) => finder.find(dir));
};
