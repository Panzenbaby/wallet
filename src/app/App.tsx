import "focus-visible";

// import { TRX } from "@payvo/sdk-trx";
// import { XLM } from "@payvo/sdk-xlm";
// import { XRP } from "@payvo/sdk-xrp";
// import { ZIL } from "@payvo/sdk-zil";
import LedgerTransportNodeHID from "@ledgerhq/hw-transport-node-hid-singleton";
// import { LUNA } from "@payvo/sdk-luna";
// import { NANO } from "@payvo/sdk-nano";
// import { NEO } from "@payvo/sdk-neo";
import { Environment } from "@payvo/profiles";
// import { ADA } from "@payvo/sdk-ada";
import { ARK } from "@payvo/sdk-ark";
// import { ATOM } from "@payvo/sdk-atom";
// import { AVAX } from "@payvo/sdk-avax";
// import { BTC } from "@payvo/sdk-btc";
// import { DOT } from "@payvo/sdk-dot";
// import { EGLD } from "@payvo/sdk-egld";
// import { ETH } from "@payvo/sdk-eth";
import { LSK } from "@payvo/sdk-lsk";
import { Offline } from "domains/error/pages";
import { Splash } from "domains/splash/pages";
import { usePluginManagerContext } from "plugins/context/PluginManagerProvider";
import React, { useLayoutEffect, useState } from "react";
import { useErrorHandler } from "react-error-boundary";
import { I18nextProvider } from "react-i18next";
import { useHistory } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import { StubStorage } from "tests/mocks";
import { setThemeSource, shouldUseDarkColors } from "utils/electron-utils";
import { bootEnvWithProfileFixtures, isE2E, isUnit } from "utils/test-helpers";

import { middlewares, RouterView, routes } from "../router";
import { ConfigurationProvider, EnvironmentProvider, LedgerProvider, useEnvironmentContext } from "./contexts";
import { useDeeplink, useNetworkStatus, useProfileSynchronizer } from "./hooks";
import { i18n as index18n } from "./i18n";
import { PluginProviders } from "./PluginProviders";
import { SentryProvider } from "./sentry/SentryProvider";
import { SentryRouterWrapper } from "./sentry/SentryRouterWrapper";
import { httpClient } from "./services";

const RouteWrappers = ({ children }: { children: React.ReactNode }) => (
	<SentryRouterWrapper>{children}</SentryRouterWrapper>
);

const Main = () => {
	const [showSplash, setShowSplash] = useState(true);
	const { env } = useEnvironmentContext();
	const { loadPlugins } = usePluginManagerContext();
	const isOnline = useNetworkStatus();
	const history = useHistory();

	useProfileSynchronizer({
		onProfileRestoreError: () => history.push("/"),
	});

	useDeeplink();

	useLayoutEffect(() => {
		setThemeSource("system");

		document.body.classList.remove(`theme-${shouldUseDarkColors() ? "light" : "dark"}`);
		document.body.classList.add(`theme-${shouldUseDarkColors() ? "dark" : "light"}`);
	}, []);

	const handleError = useErrorHandler();

	useLayoutEffect(() => {
		const boot = async () => {
			try {
				/* istanbul ignore next */
				if (isE2E() || isUnit()) {
					await bootEnvWithProfileFixtures({ env, shouldRestoreDefaultProfile: isUnit() });

					loadPlugins();
					setShowSplash(false);
					return;
				}

				/* istanbul ignore next */
				await env.verify();
				/* istanbul ignore next */
				await env.boot();
				/* istanbul ignore next */
				await loadPlugins();
			} catch (error) {
				handleError(error);
			}

			setShowSplash(false);
		};

		boot();
	}, [env, handleError, loadPlugins]);

	const renderContent = () => {
		if (showSplash) {
			return <Splash />;
		}

		if (!isOnline) {
			return <Offline />;
		}

		return <RouterView routes={routes} middlewares={middlewares} wrapper={RouteWrappers} />;
	};

	return (
		<main data-testid="Main">
			<ToastContainer closeOnClick={false} newestOnTop />

			{renderContent()}
		</main>
	);
};

export const App = () => {
	/**
	 * Ensure that the Environment object will not be recreated when the state changes,
	 * as the data is stored in memory by the `DataRepository`.
	 */

	/* istanbul ignore next */
	const storage = isE2E() || isUnit() ? new StubStorage() : "indexeddb";

	const [environment] = useState(
		() =>
			new Environment({
				coins: {
					// ADA,
					ARK,
					// ATOM,
					// AVAX,
					// BTC,
					// DOT,
					// ETH,
					// EGLD,
					LSK,
					// NEO,
					// NANO,
					// LUNA,
					// TRX,
					// XLM,
					// XRP,
					// ZIL,
				},
				httpClient,
				storage,
			}),
	);

	return (
		<I18nextProvider i18n={index18n}>
			<EnvironmentProvider env={environment}>
				<ConfigurationProvider defaultConfiguration={{ profileIsSyncingExchangeRates: true }}>
					<SentryProvider>
						<LedgerProvider transport={LedgerTransportNodeHID}>
							<PluginProviders>
								<Main />
							</PluginProviders>
						</LedgerProvider>
					</SentryProvider>
				</ConfigurationProvider>
			</EnvironmentProvider>
		</I18nextProvider>
	);
};
