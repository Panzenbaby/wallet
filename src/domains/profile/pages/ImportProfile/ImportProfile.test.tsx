/* eslint-disable @typescript-eslint/require-await */
import fs from "fs";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";
import React from "react";

import { EnvironmentProvider } from "@/app/contexts";
import { ImportProfile } from "@/domains/profile/pages/ImportProfile/ImportProfile";
import { env, fireEvent, render, screen, waitFor } from "@/utils/testing-library";

const passwordProtectedDwe = fs.readFileSync("src/tests/fixtures/profile/import/password-protected-profile.dwe");
const corruptedDwe = fs.readFileSync("src/tests/fixtures/profile/import/corrupted-profile.dwe");
const legacyJson = fs.readFileSync("src/tests/fixtures/profile/import/legacy-profile.json");
const darkThemeDwe = fs.readFileSync("src/tests/fixtures/profile/import/profile-dark-theme.dwe");
const lightThemeDwe = fs.readFileSync("src/tests/fixtures/profile/import/profile-light-theme.dwe");
const history = createMemoryHistory();

describe("ImportProfile", () => {
	let consoleSpy: jest.SpyInstance;
	let fsMock: jest.SpyInstance;

	beforeAll(() => {
		consoleSpy = jest.spyOn(console, "log").mockImplementation(() => void 0);
		fsMock = jest.spyOn(fs, "readFileSync").mockReturnValue(passwordProtectedDwe);
	});

	afterAll(() => {
		consoleSpy.mockRestore();
		fsMock.mockRestore();
	});

	it("should render first step", async () => {
		const history = createMemoryHistory();
		history.push("/profiles/import");

		const { container } = render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it("should go back", async () => {
		history.push("/profiles/import");
		const historyMock = jest.spyOn(history, "push").mockReturnValue();

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("SelectFileStep__back"));

		await waitFor(() => expect(historyMock).toHaveBeenCalledWith("/"));
		historyMock.mockRestore();
	});

	it("should change file format", async () => {
		history.push("/profiles/import");

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("SelectFileStep__change-file"));

		await waitFor(() => expect(screen.queryByTestId("SelectFileStep__change-file")).not.toBeInTheDocument());
	});

	it("should select file and go to step 2", async () => {
		history.push("/profiles/import");

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile-export.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
	});

	it("should request and set password for importing password protected profile", async () => {
		history.push("/profiles/import");

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile-export.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "S3cUrePa$sword");

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId("PasswordModal__submit-button")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("CreateProfile__form")).resolves.toBeVisible();
	});

	it("should close password modal and go back to select file", async () => {
		history.push("/profiles/import");

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile-export.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("modal__close-btn"));

		await expect(screen.findByTestId("SelectFileStep__change-file")).resolves.toBeVisible();
	});

	it("should successfully import profile and return to home screen", async () => {
		history.push("/profiles/import");
		const historyMock = jest.spyOn(history, "push").mockReturnValue();
		jest.spyOn(fs, "readFileSync").mockReturnValue(passwordProtectedDwe);

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "S3cUrePa$sword");
		await waitFor(() => expect(screen.getByTestId("PasswordModal__input")).toHaveValue("S3cUrePa$sword"));

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("CreateProfile__form")).resolves.toBeVisible();

		expect(screen.queryByTestId("InputPassword")).not.toBeInTheDocument();

		userEvent.click(screen.getByTestId("CreateProfile__submit-button"));

		await waitFor(() => expect(historyMock).toHaveBeenCalledWith("/"));
	});

	it("should successfully import legacy profile and return to home screen", async () => {
		history.push("/profiles/import");
		const historyMock = jest.spyOn(history, "push").mockReturnValue();
		jest.spyOn(fs, "readFileSync").mockReturnValueOnce(legacyJson);

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("SelectFileStep__change-file"));

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "legacy-profile.json", path: "path/to/legacy-profile.json" }],
			},
		});

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("CreateProfile__form")).resolves.toBeVisible();

		userEvent.paste(screen.getAllByTestId("Input")[0], "legacy profile");
		await waitFor(() => expect(screen.getAllByTestId("Input")[0]).toHaveValue("legacy profile"));

		userEvent.click(screen.getByTestId("CreateProfile__submit-button"));

		await waitFor(() => expect(historyMock).toHaveBeenCalledWith("/"));
	});

	it.each([
		["dark", darkThemeDwe],
		["light", lightThemeDwe],
	])("should apply theme setting of imported profile regardless of OS preferences", async (theme, dweFile) => {
		history.push("/profiles/import");

		jest.spyOn(fs, "readFileSync").mockReturnValueOnce(dweFile);

		const { asFragment } = render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("CreateProfile__form")).resolves.toBeVisible();

		expect(screen.getByRole("checkbox")).toHaveAttribute("name", "isDarkMode");

		if (theme === "dark") {
			expect(screen.getByRole("checkbox")).toBeChecked();
		} else {
			expect(screen.getByRole("checkbox")).not.toBeChecked();
		}

		expect(asFragment()).toMatchSnapshot();
	});

	it("should go to step 3 and back", async () => {
		history.push("/profiles/import");

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "S3cUrePa$sword");

		await waitFor(() => {
			expect(screen.getByTestId("PasswordModal__input")).toHaveValue("S3cUrePa$sword");
		});

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		expect(screen.getByTestId("ProcessingImport")).toBeVisible();

		await expect(screen.findByTestId("CreateProfile__form")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("CreateProfile__back-button"));

		await expect(screen.findByTestId("SelectFileStep__change-file")).resolves.toBeVisible();
	});

	it("should fail profile import and show error step", async () => {
		history.push("/profiles/import");
		const corruptedDweMock = jest.spyOn(fs, "readFileSync").mockReturnValue(corruptedDwe);

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile-export.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "wrong password");

		await waitFor(() => {
			expect(screen.getByTestId("PasswordModal__input")).toHaveValue("wrong password");
		});

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		await expect(screen.findByTestId("ImportError")).resolves.toBeVisible();

		corruptedDweMock.mockRestore();
	});

	it("should fail profile import and retry", async () => {
		history.push("/profiles/import");
		const corruptedDweMock = jest.spyOn(fs, "readFileSync").mockReturnValue(corruptedDwe);

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "wrong password");

		await waitFor(() => {
			expect(screen.getByTestId("PasswordModal__input")).toHaveValue("wrong password");
		});

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		await expect(screen.findByTestId("ImportError")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("ImportError__retry"));

		await expect(screen.findByTestId("ImportError")).resolves.toBeVisible();

		corruptedDweMock.mockRestore();
	});

	it("should fail profile import and go back to home screen", async () => {
		history.push("/profiles/import");
		const corruptedDweMock = jest.spyOn(fs, "readFileSync").mockReturnValue(corruptedDwe);
		const historyMock = jest.spyOn(history, "push").mockReturnValue();

		render(
			<EnvironmentProvider env={env}>
				<ImportProfile />
			</EnvironmentProvider>,
			{ history },
		);

		expect(screen.getByTestId("SelectFileStep__change-file")).toBeInTheDocument();
		expect(screen.getByTestId("SelectFileStep__back")).toBeInTheDocument();

		fireEvent.drop(screen.getByTestId("SelectFile__browse-files"), {
			dataTransfer: {
				files: [{ name: "profile-export.dwe", path: "path/to/sample-export.dwe" }],
			},
		});

		await expect(screen.findByTestId("ProcessingImport")).resolves.toBeVisible();
		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "wrong password");

		await waitFor(() => {
			expect(screen.getByTestId("PasswordModal__input")).toHaveValue("wrong password");
		});

		userEvent.click(screen.getByTestId("PasswordModal__submit-button"));

		await expect(screen.findByTestId("ImportError")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("ImportError__back"));

		await expect(screen.findByTestId("ImportError")).resolves.toBeVisible();

		await waitFor(() => expect(historyMock).toHaveBeenCalledWith("/"));

		historyMock.mockRestore();
		corruptedDweMock.mockRestore();
	});
});
