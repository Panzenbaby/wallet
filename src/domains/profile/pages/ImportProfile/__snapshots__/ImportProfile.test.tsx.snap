// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportProfile should render first step 1`] = `
<div>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBar__NavWrapper-sc-1nkmx4c-0 kwMnUP"
    >
      <div
        class="flex relative h-21"
      >
        <div
          class="flex flex-1 px-8 ml-12"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="NavigationBar__LogoContainer-sc-1nkmx4c-1 bfIa-di"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
            <span
              class="text-2xl font-bold"
            >
              Payvo Wallet
            </span>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    >
      <div
        class="Section__SectionWrapper-sc-plnv9l-0 efrBqv flex flex-col"
      >
        <div
          class="container px-10 mx-auto"
        >
          <div
            class="mx-auto max-w-xl"
          >
            <ul
              class="StepIndicator__StepWrapper-sc-1kbqtlh-1 htQxZl"
            >
              <li
                class="StepIndicator__StepStyled-sc-1kbqtlh-0 kcRpjv"
              />
              <li
                class="StepIndicator__StepStyled-sc-1kbqtlh-0 cqvTyD"
              />
              <li
                class="StepIndicator__StepStyled-sc-1kbqtlh-0 cqvTyD"
              />
            </ul>
            <div
              class="mt-8"
            >
              <div
                data-testid="tab-pabel__active-panel"
              >
                <div
                  class="mx-auto max-w-xl"
                >
                  <div
                    class="flex justify-between items-end bg-theme-background"
                  >
                    <div
                      class="space-y-4"
                    >
                      <h1
                        class="mb-0 text-2xl"
                        data-testid="header__title"
                      >
                        Import Profile
                      </h1>
                      <div
                        class="text-theme-secondary-text"
                        data-testid="header__subtitle"
                      >
                        Select a .dwe file with your Profile and related properties to start the import process.
                      </div>
                    </div>
                  </div>
                  <div
                    class="p-2 mt-8 h-52 rounded-lg border-2 border-dashed border-theme-secondary-300 dark:border-theme-secondary-800"
                    data-testid="SelectFile"
                  >
                    <div
                      class="h-full flex flex-col items-center justify-center rounded-lg transition-colors duration-200 bg-theme-primary-50 dark:bg-theme-secondary-800"
                      data-testid="SelectFile__drop-zone"
                    >
                      <div
                        class="sc-bdnxRM cTrAIK"
                        height="40"
                        width="40"
                      >
                        <svg>
                          dwe-file.svg
                        </svg>
                      </div>
                      <div
                        class="mt-8 space-x-px"
                      >
                        <span
                          class="font-semibold"
                        >
                          Drag & Drop or
                           
                        </span>
                        <button
                          class="relative font-semibold cursor-pointer focus:outline-none link group"
                          data-testid="SelectFile__browse-files"
                          title="Click here to upload"
                          type="button"
                        >
                          <div
                            class="absolute inset-0 -m-1 rounded group-focus-visible group-focus:ring-2 ring-theme-primary-400"
                          />
                          Browse Files
                        </button>
                      </div>
                      <div
                        class="mt-2 text-sm font-semibold text-theme-secondary-500"
                      >
                        Supported format is .dwe
                      </div>
                    </div>
                  </div>
                  <p
                    class="mt-6 text-base text-center text-theme-secondary-text"
                  >
                    <span>
                      Importing from Desktop Wallet v2?
                       
                    </span>
                    <button
                      class="relative font-semibold cursor-pointer focus:outline-none link group"
                      data-testid="SelectFileStep__change-file"
                      title="Click here."
                      type="button"
                    >
                      <div
                        class="absolute inset-0 -m-1 rounded group-focus-visible group-focus:ring-2 ring-theme-primary-400"
                      />
                      Click here.
                    </button>
                  </p>
                  <div
                    class="flex justify-end mt-8 space-x-3"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 jfKZOY"
                      data-testid="SelectFileStep__back"
                      type="button"
                    >
                      <div
                        class="flex relative items-center space-x-2"
                      >
                        Back
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
